kernel/aster-nix/src/net/socket/unix/stream/socket.rs

```diff
@@ -121,16 +132,60 @@ impl FileLike for UnixStreamSocket {
     }
 
     fn poll(&self, mask: IoEvents, poller: Option<&Poller>) -> IoEvents {
-        let inner = self.0.read();
+        let inner = self.state.read();
         match &*inner {
             State::Init(init) => init.poll(mask, poller),
             State::Listen(listen) => listen.poll(mask, poller),
             State::Connected(connected) => connected.poll(mask, poller),
         }
     }
 
+    fn ioctl(&self, cmd: IoctlCmd, arg: usize) -> Result<i32> {
+        match cmd {
+            IoctlCmd::FIONBIO => {
+                let is_nonblocking = read_val_from_user::<i32>(arg)? != 0;
+                let mut flags = self.status_flags();
+                flags.set(StatusFlags::O_NONBLOCK, is_nonblocking);
+                self.set_status_flags(flags)?;
+                Ok(0)
+            }
+            IoctlCmd::FIOASYNC => {
+                let is_async = read_val_from_user::<i32>(arg)? != 0;
+                let mut flags = self.status_flags();
+                // Did FASYNC state change ?
+                if flags.contains(StatusFlags::O_ASYNC).bitxor(is_async) {
+                    // TODO: send `SIGIO` signal to a process or a process group when
+                    // I/O is possible, user should call `fcntl(fd, F_SETOWN, pid)`
+                    // first to let the kernel know just whom to notify.
+                    flags.set(StatusFlags::O_ASYNC, is_async);
+                    self.set_status_flags(flags)?;
+                }
+                Ok(0)
+            }
+            _ => todo!(),
+        }
+    }
+
+    fn owner(&self) -> Result<Uid> {
+        let pid = *self.owner.read();
+        Ok(Uid::new(pid))
```

Comment:
Why use `Uid` here?  I think it should be a `Pid`.

Reply:
To match the `FileLike` trait, it requires a `Uid`.

Reply:
The owner and group are the system's discretionary access control UID and GID.
Will using the PID as the UID in this context compromise the original access control mechanism? 
You need to conduct more detailed research to confirm whether returning the PID is appropriate.